fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanWritePlatformServiceJpaRepositoryImpl.java

```diff
@@ -2359,6 +2359,9 @@ public void applyOverdueChargesForLoan(final Long loanId, Collection<OverdueLoan
                     null, null, null, loanId, null, null, null, null,null,null);
             LoanOverdueDTO overdueDTO = applyChargeToOverdueLoanInstallment(loanId, overdueInstallment.getChargeId(),
                     overdueInstallment.getPeriodNumber(), command, loan, existingTransactionIds, existingReversedTransactionIds);
+            if(overdueDTO == null){
```

Comment:
is there really a realistic case when `overdueDTO` can be `null` ? If not, then this could hide future bugs, and it would be better to remove this and let it explode with a NullPointerException to find such bugs, instead of silently ignoring a condition which shold never happen. If it can happen, then it's fine - but I don't see how.

Reply:
```applyChargeToOverdueLoanInstallment``` won't ever return a null `overdueDTO` as the return itself is a constructor call. I can let it explode but the problem is this whole logic is inside a for loop for all loans that are overdue. Hence, if this explodes it will impact the normal flow for overdue penalty for other loans. Hence it had to be handled gracefully. I can let it explode with a `try catch block`

Reply:
> `applyChargeToOverdueLoanInstallment` won't ever return a null `overdueDTO` as the return itself is a constructor call. 

OK but then you agree that this null check is unnecessary? If as you say it's impossible for `overdueDTO` to ever be null (because `applyChargeToOverdueLoanInstallment` ends with a `return new LoanOverdueDTO`) then why add this nul lcheck as part of this PR?

Reply:
`applyChargeToOverdueLoanInstallment` is  creating a `LoanCharge` and returning `LoanOverdueDTO`. I want to avoid creating the `LoanOverdueInstallmentCharge` on certain conditions ( the amount is zero). As the function holds all the logic for loan charges and returns using a constructor, I had to `return null` to escape the logic of charge creation. 

Hence the null pointer check is due to my code changes inside `applyChargeToOverdueLoanInstallment`

Reply:
oh right, now I get it. Sorry, I was a bit slow on this one! :smile_cat: